### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by dkfellows**
Working on improving documentation.

- [x] CI building of docs (or at least the Sphinx part)
- [x] Fix errors in doc build
- [x] Add missing method docs

There's still some places where things are double-linked. Those are quite tricky to hunt down when they're in the output of autodoc.

### CODE REVIEW

**Comment by dkfellows** [comment]
That's all the remaining faults that I can fix. The ones that are left (all warnings) are due to the combination of importing the `cli` stuff from `hpcflow.sdk` into `hpcflow.app` and the CLI documentation being generated by click; the place where it's going wrong is just too deep within the nest of generated stuff.

File: hpcflow/sdk/core/loop.py [hunk 1]
```diff
@@ -169,7 +196,25 @@ def __deepcopy__(self, memo):
 
 
 class WorkflowLoop:
-    """Class to represent a Loop that is bound to a Workflow."""
+    """
+    Class to represent a :py:class:`.Loop` that is bound to a
+    ::py:class:`~hpcflow.app.Workflow`.
```
Comment (aplowman) [inline hpcflow/sdk/core/loop.py]: There is an extra colon here I think.

**Review by aplowman — CHANGES_REQUESTED** [review]
Excellent, thanks!

File: hpcflow/sdk/core/loop.py (hunk 1 above)
Comment (dkfellows) [reply hpcflow/sdk/core/loop.py]: Oops

**Review by aplowman — APPROVED** [review]

**Pull request merged**